cms/djangoapps/contentstore/courseware_index.py

```diff
@@ -40,18 +29,69 @@ def __init__(self, message, error_list):
         self.error_list = error_list
 
 
-class CoursewareSearchIndexer(object):
+class SearchIndexerBase(object):
     """
-    Class to perform indexing for courseware search from different modulestores
+    Base class to perform indexing for courseware or library search from different modulestores
     """
 
+    INDEX_NAME = None
+    DOCUMENT_TYPE = None
+    ENABLE_INDEXING_KEY = None
+
+    INDEX_EVENT = {
+        'name': None,
+        'category': None
+    }
+
     @classmethod
-    def index_course(cls, modulestore, course_key, triggered_at=None, reindex_age=REINDEX_AGE):
+    def indexing_is_enabled(cls):
+        """
+        Checks to see if the indexing feature is enabled
+        """
+        return settings.FEATURES.get(cls.ENABLE_INDEXING_KEY, False)
+
+    @classmethod
```

Comment:
Of these classmethods, are all subclasses expected to implement them? Any methods which need to be implemented by all subclasses should use the `@abstractmethod` decorator, as is used in the modulestore.